src/components/tag/TagItem.tsx

```diff
@@ -0,0 +1,53 @@
+import styled from "@emotion/styled";
+import { HTMLAttributes } from "react";
+import { Theme } from "@emotion/react";
+
+export interface TagItemProps extends HTMLAttributes<HTMLDivElement> {
+  textColor?: keyof Theme["color"];
+  backColor?: keyof Theme["color"];
+  outLined?: boolean;
+  icon?: React.ReactNode;
+}
+
+const TagItem: React.FC<TagItemProps> = ({
+  children,
+  icon,
+  textColor = "white",
+  backColor = "primary600",
+  outLined = false,
+  ...props
+}: TagItemProps) => {
+  return (
+    <EmotionWrapper {...props} textColor={textColor} backColor={backColor} outLined={outLined}>
+      {icon}
+      {children}
+    </EmotionWrapper>
```

Comment:
오 넵 `deletable` 이라면 어떤 형태로든 이게 클린하면 지워질 수 있다는 피드백이 있어야 할 것 같습니다. 그런데 hover 시 추가하면 모바일은 어떻게 피드백을 주죠? 그냥 deletable 이면 x 를 항시 띄워놓는 것은 어떤가요..?

Reply:
오 x 표시를 띄워놓는 것도 좋겠네요! 그렇게 수정하겠습니다.

Reply:
오 저도 hover시 말고 상시 띄우는 걸로 생각하기는 했습니다 ㅋㅋㅋㅋ 모바일 생각하면 최대한 hover에서 큰(?) 피드백을 주는 일은 줄여야 할 거 같아요 (툴팁은 어쩌지)

Reply:
오 넵 툴팁... 툴팁은 좀 그렇죠.. ㅎ

hover action 을 사용할 수 없는 경우에 대한 대안을 chat GPT 가 제시해주었습니다!

## Q. 


So, in devices with mouse cursors, it is not a problem for users to 'hover' over such an element, and get the tooltip to show up, but what about mobile touchscreen users, where users don't have any chance to actually "hover" over an element before clicking on it?


## A. 


You're absolutely right. Hover interactions, such as tooltips, that rely on mouse cursor hover are not directly applicable to mobile touchscreen devices. Mobile users don't have the concept of hovering since they interact with the screen using touch gestures. However, designers have found alternative ways to provide tooltip-like information and interactions on mobile devices:

* Tap-and-Hold: On touchscreen devices, users can tap and hold on an element to trigger a tooltip-like interaction. After a brief delay, a popup or tooltip could appear, providing additional information about the element. This action simulates the hover behavior of a mouse cursor.

* Tappable Icons: Instead of relying on tooltips for information, designers can opt for using tappable icons or buttons that are designed to look and behave like buttons. Tapping these icons could reveal more details or open a modal with additional information.

* Gesture-Based Actions: Designers can incorporate gestures like swipe or pinch to trigger interactions that reveal hidden information or bring up tooltips. These gestures can be used in conjunction with icons or elements to provide context and additional details.

* Expandable Elements: Some UI elements on mobile can be designed with expandable sections. Tapping an element could expand it to reveal more content or information.

* Long Press with Context Menus: A long press on an element could trigger a context menu that provides options and additional information related to that element.

* Accordion or Tabs: In cases where tooltips are used to provide more detailed information, designers can use accordions or tabs to organize content. Tapping on an accordion item or tab header reveals the associated details.

* Info Icons: Placing small "info" icons next to elements can signal to users that there's additional information available. Tapping the icon could trigger a popup or modal with the relevant details.

* Contextual Help: Instead of relying solely on tooltips, designers can integrate contextual help within the interface itself. This could involve incorporating help sections or tutorials that users can access when needed.

It's important to note that the approach chosen should align with the overall user experience and the specific context of the application. Mobile design often requires different considerations than desktop design due to the absence of hovering and the different ways users interact with the interface. Therefore, designers need to adapt their strategies to ensure that users on touchscreen devices can access the necessary information without frustration.

Reply:
deletable인 경우 x 표시 상시 띄우도록 수정했습니다. @jaychang99 @hoon5083